src/buf/buf.rs

```diff
@@ -715,7 +720,7 @@ pub trait Buf {
     ///
     /// This function panics if there is not enough remaining data in `self`.
     fn get_f32(&mut self) -> f32 {
-        buf_get_impl!(self, 4, BigEndian::read_f32);
+        unsafe { *(&Self::get_u32(self) as *const u32 as *const f32) }
```

Comment:
Perhaps you can get rid of these unsafe block.

https://github.com/rust-lang/rust/issues/60446
> unlike corresponding integer methods, these can be implemented without unsafe code. So the need for having them in the standard library feels less pressing. At the same time, it might not be obvious to everyone that to_bits and from_bits can be combined with integer methods in this way.

Reply:
Yep, you're right